traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/router/TrafficRouter.java

```diff
@@ -262,7 +264,8 @@ public Geolocation getLocation(final String clientIP, final DeliveryService deli
 			}
 		} else {
 			// Deep caching not enabled for this Delivery Service; use the regular CZ
-			cacheLocation = getCoverageZoneCacheLocation(request.getClientIP(), ds);
+			cacheLocation = getCoverageZoneCacheLocation(request.getClientIP(), ds, track, useDeep);
+			networkNode = getNetworkNode(request.getClientIP());
```

Comment:
Generally I'm not a big fan of polluting this method with extra NetworkNode stuff, especially since `getNetworkNode` is already called in `getCoverageZoneCacheLocation`. I think ideally `getCoverageZoneCacheLocation` should return us the information we need.

Since we're passing in the `Track` object, we could probably add that information there, ie something like `track.noBackupLocationsAvailable()` which would get set within `getCoverageZoneCacheLocation`. Then we can update the logic starting on L277 to something like this:
```
if (track.noBackupLocationsAvailable() or ds.isCoverageZoneOnly()) { 
    if (ds.geoRedirect) {...}
    else {
        resultDetails = DS_CZ_ONLY if ds.isCoverageZoneOnly else DS_CZ_NO_BKP
        ...
    }
}
```